pallets/settlement/src/lib.rs

```diff
@@ -1364,29 +1331,44 @@ impl<T: Trait> Module<T> {
         }
     }
 
-    fn is_instruction_executed(
+    /// Schedule a given instruction to be executed on the next block only if the
+    /// settlement type is `SettleOnAffirmation` and no. of affirms pending is 0.
+    fn restricted_schedule_instruction(
         affirms_pending: u64,
         settlement_type: SettlementType<T::BlockNumber>,
         id: u64,
-    ) -> DispatchResultWithPostInfo {
-        let execute_instruction_result =
-            if affirms_pending == 0 && settlement_type == SettlementType::SettleOnAffirmation {
-                Self::execute_instruction(id).1
-            } else {
-                Ok(PostDispatchInfo {
-                    actual_weight: Some(Zero::zero()),
-                    pays_fee: Default::default(),
-                })
-            };
-        execute_instruction_result
+    ) -> DispatchResult {
+        if affirms_pending == 0 && settlement_type == SettlementType::SettleOnAffirmation {
+            // Schedule instruction to be executed in the next block.
+            let execution_at = system::Module::<T>::block_number() + One::one();
+            Self::schedule_instruction(id, execution_at)?;
+        }
+        Ok(())
+    }
+
+    /// Schedule execution of given instruction at given block number.
+    ///
+    /// NB - It is expected to execute the given instruction into the given block number but
+    /// it is not a guaranteed behavior, Scheduler may have other high priority task scheduled
+    /// for the given block so there are chances where the instruction execution block no. may drift.
+    fn schedule_instruction(instruction_id: u64, execution_at: T::BlockNumber) -> DispatchResult {
+        let call = Call::<T>::execute_scheduled_instruction(instruction_id).into();
+        T::Scheduler::schedule(
+            DispatchTime::At(execution_at),
+            None,
+            LOWEST_PRIORITY,
```

Comment:
Instead of setting it as the lowest priority, let's set it as something between 100 and 200. Settlements are not critical but they aren't the lowest priority either. Perhaps define all priorities in a common file.